pkg/synopsysctl/cmd_update.go

```diff
@@ -486,150 +186,100 @@ var updateBlackDuckCmd = &cobra.Command{
-					if err != nil {
-						return errors.Wrap(err, "failed to list pods to stop BlackDuck for setting group ownership")
-					}
-					if len(pods.Items) == 0 {
-						break
-					}
-					time.Sleep(time.Second * 5)
-					waitCount = waitCount + 1
-					if waitCount%5 == 0 {
-						log.Debugf("waiting for Black Duck to stop - %d pods remaining", len(pods.Items))
+			secrets, err := blackduck.GetCertsFromFlagsAndSetHelmValue(args[0], namespace, cmd.Flags(), helmValuesMap)
+			if err != nil {
+				return err
+			}
+			for _, v := range secrets {
+				if _, err := kubeClient.CoreV1().Secrets(namespace).Create(&v); err != nil {
+					if k8serrors.IsAlreadyExists(err) {
+						if _, err := kubeClient.CoreV1().Secrets(namespace).Update(&v); err != nil {
+							return fmt.Errorf("failed to update certificate secret: %+v", err)
+						}
+					} else {
+						return fmt.Errorf("failed to create certificate secret: %+v", err)
 					}
 				}
 			}
-			// Get a list of Persistent Volumes based on Persistent Volume Claims
-			pvcList, err := util.ListPVCs(kubeClient, blackDuckNamespace, fmt.Sprintf("app=blackduck,component=pvc,name=%s", blackDuckName))
-			if err != nil {
-				return errors.Wrap(err, fmt.Sprintf("failed to list PVCs to update the group ownership"))
+
+			var extraFiles []string
+			size, found := instance.Config["size"]
+			if found {
+				extraFiles = append(extraFiles, fmt.Sprintf("%s.yaml", size.(string)))
 			}
-			// Map the Persistent Volume to the respective Security Context file Ownership value - if security contexts are not provided then this map will be empty
-			pvcNameToFileOwnershipMap := map[string]int64{}
-			pvcNameToSecurityContextNameMap := map[string]string{
-				"blackduck-postgres":         "blackduck-postgres",
-				"blackduck-cfssl":            "blackduck-cfssl",
-				"blackduck-registration":     "blackduck-registration",
-				"blackduck-zookeeper":        "blackduck-zookeeper",
-				"blackduck-authentication":   "blackduck-authentication",
-				"blackduck-webapp":           "blackduck-webapp",
-				"blackduck-logstash":         "blackduck-webapp",
-				"blackduck-uploadcache-data": "blackduck-uploadcache",
+
+			if err := util.UpdateWithHelm3(args[0], namespace, blackduckChartRepository, helmValuesMap, kubeConfigPath, extraFiles...); err != nil {
+				return err
 			}
-			for _, pvc := range pvcList.Items {
-				r, _ := regexp.Compile("blackduck-.*")
-				pvcNameKey := r.FindString(pvc.Name) // removes the "<blackduckName>-" from the PvcName
-				sc := appsutil.GetSecurityContext(currBlackDuck.Spec.SecurityContexts, pvcNameToSecurityContextNameMap[pvcNameKey])
-				if sc != nil {
-					if sc.RunAsUser != nil {
-						pvcNameToFileOwnershipMap[pvc.Name] = *sc.RunAsUser
-					}
-				}
+		} else if isOperatorBased {
+			ok, err := util.IsVersionGreaterThanOrEqualTo(cmd.Flag("version").Value.String(), 2019, time.April, 0)
```

Comment:
What happens if cmd.Flag("version") is not passed into the update command? 
Theoretically this code should only be executed if the user is "upgrading" an old BlackDuck version to a new one so it must have been set. But maybe we should put a check in for if a customer accidentally tries to "update" a different field of an old BlackDuck.

Reply:
This is what I put in Alert:
```
if !isOperatorBased && instance != nil {
	err = updateAlertHelmBased(cmd, alertName)
} else if isOperatorBased {
	// user must be upgrading to a new version
	if !cmd.Flag("version").Changed { // TODO fill in the alert version
		return fmt.Errorf("you must upgrade this Alert version with --version to use this synopsysctl binary - modifying Alert versions before XXXXX are not supported with this binary")
	}
	err = updateAlertOperatorBased()
}
```

Reply:
Good catch!